source/scripts/boot.lua

```diff
@@ -357,80 +335,114 @@ function love.errhand(message)
         end
     end
 
-    local joystick = love.joystick.getJoysticks()[1]
-
     local quit_string = "start"
     if love._console_name == "Switch" then
         quit_string = "plus"
     end
 
-    love.draw = function()
-        draw()
-    end
-
     love.gamepadpressed = function(joy, button)
         if button == quit_string then
             love.event.quit()
         end
     end
 end
 
-local function pseudoRequireConf()
-    return require('conf')
-end
+-- love.errhand was deprecated in LOVE 11.0, replaced by love.errorhandler.
+love.errhand = love.errorhandler
 
-local confSuccess
-if love.filesystem.getInfo("conf.lua") then
-    confSuccess = xpcall(pseudoRequireConf, love.errhand)
+function love.threaderror(t, err)
+	error("Thread error ("..tostring(t)..")\n\n"..err, 0)
+end
 
-    if not confSuccess then
-        return
-    end
+--################--
+-- BOOT THE GAME! --
+--################--
 
-    if love.conf then
-        local function confWrap()
-            love.conf(config)
-        end
+function love.boot()
+    -- We can't throw any errors just yet because we want to see if we can
+    -- load and use conf.lua in case the user doesn't want to use certain
+    -- modules, but we also can't error because graphics haven't been loaded.
+    local confok, conferr
 
-        confSuccess = xpcall(confWrap, love.errhand)
+    if love.filesystem.getInfo("conf.lua", "file") then
+        confok, conferr = pcall(require, "conf")
 
-        if not confSuccess then
-            return
+        if confok and love.conf then
+            confok, conferr = pcall(love.conf, config)
         end
     end
-end
 
-love.filesystem.setIdentity(config.identity)
+    -- Set the game identity for saving.
+    love.filesystem.setIdentity(config.identity)
 
-local function pseudoRequireMain()
-    return require("main")
-end
+    -- Load the modules.
+    local modules =
+    {
+        -- Try to load love.graphics, window, and event first in case we need
+        -- to jump to errhand after it. (window is only required on Switch.)
+        "graphics",
+        "window",
+        "event",
+        "timer",
+        "joystick",
+        "audio",
+        "keyboard",
+        "math",
+        "system",
+    }
+    for i, v in ipairs(modules) do
+        if config.modules[v] then
+            love[v] = require("love." .. v)
+        end
+    end
 
-if love.filesystem.getInfo("main.lua") then
-    --Try main
-    local result = xpcall(pseudoRequireMain, love.errhand)
-    if not result then
-        return
+    -- Load any Switch exclusive modules.

```

Comment:
This should be `love._console_name` instead of `love._os[2]`.
Another way to do this is just insert to the main modules table `{“image”, “thread”}` and *then* iterate the array and require each one.

Edit: also “touch” is common between consoles now.